### PR TITLE
Resource Status beta promotion

### PR DESCRIPTION
**Pull request by litong01**
Signed-off-by: Tong Li <litong01@us.ibm.com>

### CODE REVIEW

File: features/resource_status.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+- [x] End-to-end tests cover samples/tutorials
+- [x] Fixed issues have tests to prevent regressions
+- [x] Stability/stress test suite includes coverage for the feature.
+
+    https://github.com/istio/istio/issues/30179
+
+**Performance**
+
+- [x] Feature coverage and test plans written and approved 
+- [x] Tests exist with the feature enabled that can be integrated with our automated performance testing.
```
Comment (howardjohn) [inline features/resource_status.md:165]: Can you expand on this? Where is the plan and tests?

File: features/resource_status.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+- [ ] Upgrade notes have been added. 
+
+**Tests**
+
+- [x] Integration tests cover feature edge cases
+- [x] End-to-end tests cover samples/tutorials
+- [x] Fixed issues have tests to prevent regressions
+- [x] Stability/stress test suite includes coverage for the feature.
+
+    https://github.com/istio/istio/issues/30179
```
Comment (howardjohn) [inline features/resource_status.md:160]: Its not clear how this umbrella issue gives proof to the above section

File: features/resource_status.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+**Design**
+
+- [x] Design doc describing the intention of the feature, how it will be
+	implemented, and any thoughts on how to test the feature has been approved by
+	relevant work group leads
+- [x] Feature coverage and test plans written and approved.
+
+**Docs** 
+
+- [x] Documentation on istio.io includes performance expectations; may have caveats.
```
Comment (howardjohn) [inline features/resource_status.md:146]: Can you include a link?

File: features/resource_status.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+---
+
+## Beta
+
+### Requirements: 
+
+**Design**
+
+- [x] Design doc describing the intention of the feature, how it will be
```
Comment (howardjohn) [inline features/resource_status.md:139]: Can you include a link?

File: features/resource_status.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+- [ ] TOC has reviewed the API and determined it to be complete. 
+
+**Tooling**
+
+- [ ] Any necessary tooling to use/debug the feature has been implemented and is complete. 
+
+**Bugs**
+
+- [x] Feature has no known major issues.
+    Resolved issues: https://github.com/istio/istio/issues/30179
```
Comment (howardjohn) [inline features/resource_status.md:178]: I am not sure these are all solved..? Seems a lot of them were just closed for being stale

Comment (litong01) [reply features/resource_status.md:178]: @howardjohn John, thanks for looking at this, I am hoping that this PR https://github.com/istio/istio.io/pull/11121 can be merged into the docs which contains a link to the tests that I have done. The tests and results are here https://github.com/litong01/multicluster-istio/blob/main/perf/charts/README.md

Comment (litong01) [reply features/resource_status.md:178]: @howardjohn I think we've talked about this list awhile back, the one that needs to be resolved is this https://github.com/istio/istio/issues/30200. I've merged one PR which introduced this new parameter ISTIOD_ANALYSIS_INTERVAL to allow a user to control how often to run analyzers. Unless there is any other issue need to be resolved. If you do have other issues wanting to be resolved, I'm happy to look into.

File: features/resource_status.md (hunk 4 above)
Comment (litong01) [reply features/resource_status.md:139]: https://docs.google.com/document/d/1OipeJ1NnBAGkw27CSL02eykbhF8zRdWKBfwtrqQOhNE/edit#heading=h.qex63c29z2to

@howardjohn this is the doc that was generated long time ago, I did some updates recently.

Comment (litong01) [reply features/resource_status.md:139]: The link is also in this PR.

File: features/resource_status.md (hunk 5 above)
Comment (howardjohn) [reply features/resource_status.md:178]: I don't think that solves the problem... you cannot reasonably set a high value (say, 10m), since then the analysis info is too outdated. But you cannot set it low (10s), since then it is way to expensive.

Its not clear why we should have code doing a TON of work looking at the exact same config when it isn't changed. All other parts of Istio are reactive (eg when VS changes, push the new VS). This means it is faster and cheaper.

---

* https://github.com/istio/istio/issues/29709
* https://github.com/istio/istio/issues/30512
* https://github.com/istio/istio/issues/30362
* https://github.com/istio/istio/issues/30292
* https://github.com/istio/istio/issues/30126
* https://github.com/istio/istio/issues/30122
* https://github.com/istio/istio/issues/28015

Just briefly going through the list, the following were stale-closed, so I assume they aren't fixed. Anecdotally, I enabled this flag in my cluster and was immediately presenting with 1000s of error messages in Istiod logs, so it seems at least some must be unfixed.

File: features/resource_status.md (hunk 4 above)
Comment (howardjohn) [reply features/resource_status.md:139]: That is not a proper design doc, that only describes the schema. My concerns are not with the schema but everything else

File: features/resource_status.md (hunk 1 above)
Comment (litong01) [reply features/resource_status.md:165]: @howardjohn this I am not really sure, I think @therealmitchconnors probably can help more on this one.

File: features/resource_status.md (hunk 4 above)
Comment (litong01) [reply features/resource_status.md:139]: @howardjohn john, you would like to see a document with some design on how this should been implemented, right? like when it will start to run, how analyzer is designed, how an analyzer gets invoked, resources being retrieved etc, are these the things that you are looking for?

File: features/resource_status.md (hunk 5 above)
Comment (litong01) [reply features/resource_status.md:178]: @howardjohn yes, based on the current design, it runs every 10 seconds to analyze all the resources in watched namespaces. that is how it was designed. unless we want to make changes to that, these error messages will be present in the logs unfortunately.

**Comment by whitneygriffith** [comment]
@litong01 @therealmitchconnors are you still working on this PR? There's still some unaddressed comments and we would like to close the PR or revive the work to get this completed.

**Comment by craigbox** [comment]
Second call on closing, @litong01 has moved on but @therealmitchconnors may have some input?
